src/geometries/ParametricGeometry.js

```diff
@@ -72,6 +77,16 @@ function ParametricBufferGeometry( func, slices, stacks ) {
 			var p = func( u, v );
 			vertices.push( p.x, p.y, p.z );
 
+			// approximate tangent plane vectors via central difference
+
+			pu.subVectors( func( u + EPS, v ), func( u - EPS, v ) );
+			pv.subVectors( func( u, v + EPS ), func( u, v - EPS ) );
```

Comment:
@Mugen87 Actually, there should be no reason to instantiate `4 * slices * stacks` `Vector3`'s when using central differences. We could add an `optionalTarget` to `func()` and reuse just two instances.
```javascript
var result = optionalTarget || new THREE.Vector3();
...
return result.set( x, y, z );
```